packages/financial-templates-lib/src/clients/FinancialContractFactoryEventClient.js

```diff
@@ -0,0 +1,109 @@
+// A thick client for getting information about FinancialContractFactory events.
+
+class FinancialContractFactoryEventClient {
+  /**
+   * @notice Constructs new FinancialContractEventClient.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} financialContractFactoryAbi ExpiringMultiPartyCreator or PerpetualMultiPartyCreator truffle ABI object.
+   * @param {Object} web3 Web3 provider from truffle instance.
+   * @param {String} financialContractFactoryAddress Ethereum address of the factory contract deployed on the current network.
+   * @param {Integer} startingBlockNumber Offset block number to index events from.
+   * @param {Integer} endingBlockNumber Termination block number to index events until. If not defined runs to `latest`.
+   * @return None or throws an Error.
+   */
+  constructor(
+    logger,
+    financialContractFactoryAbi,
+    web3,
+    financialContractFactoryAddress,
+    startingBlockNumber = 0,
+    endingBlockNumber = null,
+    contractType = "Perpetual"
```

Comment:
One quick note: we could make this the name of the factory contract (ExpiringMultiPartyCreator or PerpetualCreator). Then that could be pulled directly from the artifact. Not strongly opinionated, but that seems a bit more obvious since we call these contracts different names (EMP vs ExpiringMultiParty vs ExpiribgMultiPartyCreator).

Reply:
Definitely